app/views/products/index.html.erb

```diff
@@ -1,32 +1,83 @@
 <h1 class="text-2xl font-bold mb-4">Products</h1>
 
-  <div class="overflow-x-auto">
-    <table class="min-w-full bg-white border border-gray-200">
-      <thead>
-        <tr>
-          <th class="px-6 py-3 border-b border-gray-200 bg-gray-100 text-left text-sm font-semibold text-gray-700">Name</th>
-          <th class="px-6 py-3 border-b border-gray-200 bg-gray-100 text-left text-sm font-semibold text-gray-700">Category</th>
-          <th class="px-6 py-3 border-b border-gray-200 bg-gray-100 text-left text-sm font-semibold text-gray-700">Price</th>
-          <th class="px-6 py-3 border-b border-gray-200 bg-gray-100 text-left text-sm font-semibold text-gray-700">Actions</th>
-        </tr>
-      </thead>
-      <tbody>
-        <% @products.each do |product| %>
-          <tr class="hover:bg-gray-50">
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800 font-medium"><%= product.name %></td>
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800"><%= product.category&.name || 'N/A' %></td>
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800">$<%= product.price %></td>
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800">
-              <% if user_signed_in? %>
-                      <%= button_to 'Add to Cart', add_to_cart_order_path(current_cart, product_id: product.id), method: :post, class: 'bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded' %>
+<% if params[:category].present? %>
+  <% category = params[:category].to_i %>
+  <div class="bg-white">
+    <div class="mx-auto max-w-7xl px-4 py-16 sm:px-6 lg:px-8">
+      <div class="mt-6 grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-4">
+        <% Product.where(category_id: category).each do |product| %>
```

Comment:
Use @products here because you are already finding @products in index action in products_controller.